packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -46,10 +46,11 @@ export type GridColumnConfig<T> = (
     | { type: "staticSelect"; values?: string[] }
     | { type: "block"; block: ImportReference }
 ) & { name: keyof T } & DataGridSettings;
-export type GridConfig<T extends { __typename?: string }> = {
+export type GridConfig<T extends { __typename?: string }, Queries extends { __typename?: "Query" } = { __typename: "Query" }> = {
```

Comment:
Should the Queries type always be passed? Is that really worth it? why don't you do it in demo? And have you considered extracting the `T`type out of it?

Reply:
> Should the Queries type always be passed?

the queries-type is optional and enforces using an existing query-string. if not provided possible values for "query" config is empty. query config is optional, so if I don't need to define a custom query I don't need to provide the Query-Type. Currently it only helps preventing typos. So we don't really need the queries-type...

> why don't you do it in demo?

I'll check again if there is any query without default-naming.

> And have you considered extracting the `T`type out of it?

what do mean with `T`type? I thought about checking the return type to match the gqlType, but it seems this is more complicated, because of paginated/nodes/type-required-to-check. And this information would be handy in configuration to allow only queries with correct return type. For now my use-case only requires me to use the "clubsByParentClub" instead of "clubs"

Reply:
> my question was, why you don't pass the second argument (Queries) in demo.

because it's not required, only if you need to fetch the data from a query named differently.

> My thought was that we could eventually automatically get that type, so you don't have to specify two arguments

I looked into this type, but it's only a reference to the entity itself. There is no reference from the entity to queries returning this entity.
<img width="359" alt="image" src="https://github.com/vivid-planet/comet/assets/1324250/7839f2d5-8d48-4d22-9a5f-4c8629654d5d">
This is the type referenced in the second type. And the highlighted is the default query used to fetch data for grid. I need to generate a grid with e.g. `productsByManufacturer(manufacturer: ID!, filter: Filter) {...}` instead of `products(filter: Filter) {...}`
<img width="676" alt="image" src="https://github.com/vivid-planet/comet/assets/1324250/b822896e-9d92-46ed-8ec5-057305303e90">

Reply:
> There is no reference from the entity to queries returning this entity.

but the other way around.

Reply:
> because it's not required, only if you need to fetch the data from a query named differently.

I doubt that it will be used by anyone if you don't enforce it.